sheriff_test.go

```diff
@@ -793,7 +852,7 @@ func TestMarshal_BooleanPtrMap(t *testing.T) {
 	expect, err := json.Marshal(toMarshal)
 	assert.NoError(t, err)
 
-	assert.Equal(t, string(marshal), string(expect))
+	assert.JSONEq(t, string(marshal), string(expect))
```

Comment:
as we use a `map` as the input the order of the result cannot be guaranteed.

Using `JSONEq` will ensure the json is correct (even if unordered)